src/osd/PrimaryLogPG.cc

```diff
@@ -11343,15 +11343,13 @@ void PrimaryLogPG::on_removal(ObjectStore::Transaction *t)
   pg_log.reset_backfill();
   dirty_info = true;
 
-
   // clear log
   PGLogEntryHandler rollbacker{this, t};
   pg_log.roll_forward(&rollbacker);
 
-  write_if_dirty(*t);
```

Comment:
See PG::do_peering_event(), where all of the state machine events come come through.  Previously on_removal() was special because it was called by the ad-hoc PG deletion code in OSD.cc instead of via the state machine; now it is called via the state machine (which handles this for us).

Reply:
👍